src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -298,7 +319,8 @@ entails SolverOptions{..} constraint context hints =
               | pairwiseAny overlapping (map snd tcds) = do
                   tell . errorMessage $ OverlappingInstances className' tyArgs (tcds >>= (toList . namedInstanceIdentifier . tcdValue . snd))
                   return $ uncurry Solved (head tcds)
-              | otherwise = return $ uncurry Solved (minimumBy (compare `on` length . tcdPath . snd) tcds)
+              | otherwise =
+                return $ uncurry Solved (minimumBy (compare `on` (\d -> (tcdChain d, tcdIndex d, length (tcdPath d))) . snd) tcds)
```

Comment:
Now that we are considering instances in order, I shouldn't need to do this any more.